subworkflows/nf-core/align_bowtie2/main.nf

```diff
@@ -0,0 +1,40 @@
+//
+// Alignment with Bowtie2
+//
+
+params.align_options    = [:]
+params.samtools_options = [:]
+
+include { BOWTIE2_ALIGN     } from '../../../modules/bowtie2/align/main' addParams( options: params.align_options    )
```

Comment:
Do we include the modules, or do we have a dedicated "lock" directory with physical copies of the modules for each subworkflow?

Reply:
I think with all this CI, we should be able to just depend on the up-to-date modules. If something changes underneath, and breaks/changes an output, we'll know.

Now that I'm thinking about it, we should test everything periodically just to make sure we didn't miss anything or something is flaky. I opened #664 for this.

Reply:
> I think with all this CI, we should be able to just depend on the up-to-date modules.

There could be rare cases where a subworkflow is not compatible with the latest version of a module. But I'd risk it until we have such a case and then figure out a solution.

Reply:
I disagree - I think subworkflows should work in the same way as pipelines: pinning module versions. Otherwise different pipelines using the same version of the same subworkflow could use different module versions, which is super confusing.

Reply:
(Also, someone _will_ request this and I think it's easier to build in from the start in the tooling rather than doing loads of refactoring).

Reply:
to keep the discussion going :stuck_out_tongue: 

> someone will request this

I have the feeling this is more of a theoretical problem (the same with versioning modules - a lot of people have requested it but it seems nobody ever really needed it. I was one of the fiercest supporter of this idea in #2, but that was before I actually started *using* modules. 

> Otherwise different pipelines using the same version of the same subworkflow

In that case it could happen that a pipeline has different versions of a module (once in a subworkflow and once outside) which is at least as confusing

I am also a bit worried about the increasing overhead cost of updating/fixing something in a pipeline. With DSL1 you would have just fixed it and were done with it. With modules, you already have to first fix the module and then pull it in. With pinned versions in subworkflows you would additionally have to update the subworkflow before pulling it in. 

Moreover I'm worried about how to keep everything up-to-date. If we go with pinned versions, we need very, very good tooling and a high degree of automation.

Reply:
> I disagree - I think subworkflows should work in the same way as pipelines: pinning module versions. Otherwise different pipelines using the same version of the same subworkflow could use different module versions, which is super confusing.

I guess the issue I was thinking of was if they installed the subworkflow AND wanted to use the underlying module outside of the subworkflow. 

I see no issue with the same install in the `modules.json`(Or where ever the lock file is for subworkflows). But maybe we have a check that the underlying modules aren't installed. 

One consideration though is that a module change shouldn't break a subworkflow. The subworkflow should be updated if the module breaks it, which would trigger a release.